_posts/2021-09-02-gsoc-final-report-orion.md

```diff
@@ -0,0 +1,194 @@
+# define solute AtomGroup
+li_atoms = u.atoms.select_atoms("type 22")
+
+# define solvent AtomGroups
+PF6 = u.atoms.select_atoms("byres type 21")
+BN = u.atoms.select_atoms("byres type 5")
+FEC = u.atoms.select_atoms("byres type 19")
+```
+
+Once we have the universe, solute, and solvents defined, it is quite simple to instantiate and run a solution! Solution, like all MDA analysis classes, subclasses the [AnalysisBase] class and requires `Solution.run()` call to initiate the analysis.
+
+```python
+# instantiate solution
+from solvation_analysis.solution import Solution
+
+solution = Solution(li_atoms, {'PF6': PF6, 'BN': BN, 'FEC': FEC})
+
+solution.run()
+```
+
+Over the course of a few seconds, the solvation radii and the solvation shell of each ion are automatically identified and stored in a `pandas.DataFrame` (as described above). We can then examine how well the solvation radii were characterized by calling `solution.plot_solvation_radius('<<solvent name>>')`. For example,
+
+```python
+# we just need this to display our plot
+import matplotlib.pyplot as plt
+
+solution.plot_solvation_radius('BN')
+plt.show()
+```
+![bn plot]({{site.images}}/final_report_orion/bn_plot.png)
+
+
+Now that we have a completed solution, we can easily print the ion pairing, coordination numbers, and solvation radii.
+
+```python
+>>> solution.coordination.cn_dict
+{'BN': 4.33, 'FEC': 0.25, 'PF6': 0.12}
+>>> solution.pairing.pairing_dict
+{'BN': 1.0, 'FEC': 0.21, 'PF6': 0.12}
+>>> solution.radii
+{'PF6': 2.60, 'BN': 2.61, 'FEC': 2.43}
+```
+
+We can also use `solvation_analysis` to identify the most common solvation shell compositions and find examples of them for visualization. Below, we are using the `speciation` class to find the most common compositions of solvation shells and then filtering out all shells with a frequency less than 2%.
+
+```python
+solution.speciation.speciation_percent.query("count > 0.02")
+```
+
+![speciation dataframe]({{site.images}}/final_report_orion/speciation_frame_new.png)
+
+We can pick out a specific examples of that shell for visualization. First we find all shells with a given composition.
+
+```python
+# find all shells matching the given dictionary
+solution.speciation.find_shells({'BN': 4, 'FEC': 0, 'PF6': 1})
+```
+![selection dataframe]({{site.images}}/final_report_orion/selection_frame.png)
+
+Then we select the solvation shell and use our favorite visualization package to view it!
+
+```python
+# get the shell atoms
+atoms = solution.solvation_shell(6, 0)
+
+# visualization is not covered here. I suggest nglview!
+visualize(atoms)
+```
+
+![shell visualization]({{site.images}}/final_report_orion/shell.png)
+
+Using that approach, you can find all the common solvation shells and generate visualizations for each. For example, see this speciation plot of the BN:FEC system from the example. I've plotted the percentage of each solvation shell at two temperatures and then visualized examples of those shells in the margin.
```

Comment:
Is this percentage or probability? looks < 1%, but that could be correct IDK about the system.

Reply:
I guess its really a fraction? I corrected the verbiage here.

Reply:
@orioncohen did you also update Y axis of graph?

Reply:
Corrected!